morpheus/_lib/include/morpheus/stages/add_scores_stage_base.hpp

```diff
@@ -81,6 +88,181 @@ class AddScoresStageBase
     std::size_t m_min_col_count;
 };
 
+template <typename InputT, typename OutputT>
+AddScoresStageBase<InputT, OutputT>::AddScoresStageBase(std::map<std::size_t, std::string> idx2label,
+                                                        std::optional<float> threshold) :
+  base_t(),
+  m_idx2label(std::move(idx2label)),
+  m_threshold(threshold),
+  m_min_col_count(m_idx2label.rbegin()->first)  // Ordered map's largest key will be the last entry
+{
+    this->pipe(rxcpp::operators::map([this](sink_type_t x) {
+        return this->on_data(std::move(x));
+    }));
+}
```

Comment:
Lets switch to using [explicit template instantiation](https://stackoverflow.com/a/2351622) since there are only two supported types for this class. To do this:

1. Move all implementation details back to the cpp file.
2. Add using declarations for the supported types, i.e. `using AddScoresStageBaseMM = AddScoresStageBase<MultiResponseMessage, MultiResponseMessage>;` (You can use `// NOLINT(readability-identifier-naming)` to avoid the linter error)
3. At the end of the cpp file, explicitly instantiate the classes we want: `template class AddScoresStageBase<MultiResponseMessage, MultiResponseMessage>;`

Do this for all of the new templated classes